kernel/aster-nix/src/net/socket/ip/stream/connected.rs

```diff
@@ -68,6 +68,24 @@ impl ConnectedStream {
         self.update_io_events(pollee);
     }
 
+    pub fn register_observer(
+        &self,
+        pollee: &Pollee,
+        observer: Weak<dyn Observer<IoEvents>>,
+        mask: IoEvents,
+    ) -> Result<()> {
+        pollee.register_observer(observer, mask);
+        Ok(())
+    }
+
+    pub fn unregister_observer(
+        &self,
+        pollee: &Pollee,
+        observer: &Weak<dyn Observer<IoEvents>>,
+    ) -> Result<Weak<dyn Observer<IoEvents>>> {
+        Ok(pollee.unregister_observer(observer).unwrap())
```

Comment:
Please use `ok_or_else` instead of `unwrap`. `unwrap` can only be used if the statement will never return `None` or `Err`.

```rust
pollee.unregister_observer(observer).ok_or_else(|_| Error::with_message(Errno::EINVAL, "fails to unregister observer"))
```

Reply:
Thank you for your instruction. I have changed the statement now.